paddle/phi/kernels/onednn/matmul_kernel.cc

```diff
@@ -0,0 +1,450 @@
+};
+
+/* OT: output data type */
+template <typename XT, typename YT, typename OT>
+std::shared_ptr<MulPrimitiveFactory<XT, YT, OT>> GetPrimitiveFactory(
+    const OneDNNContext &dev_ctx,
+    const DenseTensor *input_x,
+    const DenseTensor *input_y,
+    const dnnl::engine &onednn_engine) {
+  std::string key =
+      funcs::CreateKey(dev_ctx,
+                       paddle::framework::TransToProtoVarType(input_x->dtype()),
+                       phi::vectorize(input_x->dims()),
+                       paddle::framework::TransToProtoVarType(input_y->dtype()),
+                       phi::vectorize(input_y->dims()),
+                       dev_ctx.GetOutputsName("Out")[0]);
+  key = funcs::ExtendKeyWithThreadInfoIfNeeded(dev_ctx, key);
+
+  auto prim_creator = std::static_pointer_cast<MulPrimitiveFactory<XT, YT, OT>>(
+      dev_ctx.GetBlob(key));
+
+  if (prim_creator == nullptr) {
+    prim_creator =
+        std::make_shared<MulPrimitiveFactory<XT, YT, OT>>(onednn_engine);
+    dev_ctx.SetBlob(key, prim_creator);
+  }
+
+  return prim_creator;
+}
+
+/* XT: input x data type, YT: input y data type */
+template <typename XT, typename YT, typename Context>
+inner_product_forward GetMulPrimitive(const Context &dev_ctx,
+                                      const DenseTensor *input_x,
+                                      const DenseTensor *input_y,
+                                      DenseTensor *output,
+                                      int x_num_col_dims,
+                                      int y_num_col_dims,
+                                      const dnnl::engine &onednn_engine) {
+  constexpr bool is_int8 = funcs::is_int8<XT>();
+  bool force_fp32_output =
+      dev_ctx.HasDnnAttr("force_fp32_output")
+          ? PADDLE_GET_CONST(bool, dev_ctx.GetDnnAttr("force_fp32_output"))
+          : false;
+
+  if (is_int8 && !force_fp32_output) {
+    return GetPrimitiveFactory<XT, YT, int8_t>(
+               dev_ctx, input_x, input_y, onednn_engine)
+        ->CreateMulPrimitive(
+            input_x, input_y, output, x_num_col_dims, y_num_col_dims, dev_ctx);
+
+  } else {
+    return GetPrimitiveFactory<XT, YT, float>(
+               dev_ctx, input_x, input_y, onednn_engine)
+        ->CreateMulPrimitive(
+            input_x, input_y, output, x_num_col_dims, y_num_col_dims, dev_ctx);
+  }
+}
+
+/* XT: input x data type */
+template <typename XT, typename Context>
+void MatmulWithFlattenKernelINT8(const Context &dev_ctx,
+                                 const DenseTensor &x,
+                                 const DenseTensor &y,
+                                 int x_num_col_dims,
+                                 int y_num_col_dims,
+                                 DenseTensor *out) {
+  PADDLE_ENFORCE_EQ(paddle::platform::is_cpu_place(dev_ctx.GetPlace()),
+                    true,
+                    paddle::platform::errors::PreconditionNotMet(
+                        "Operator DNNL Mul must use CPUPlace"));
```

Comment:
```suggestion
                    paddle::platform::errors::PreconditionNotMet(
                        "oneDNN MatmulWithFlatten kernel must use CPUPlace"));
```
now errors are located in `phi::errors` namespace